src/Symfony/Component/HttpKernel/Bundle/AbstractBundle.php

```diff
@@ -47,4 +47,16 @@ public function getContainerExtension(): ?ExtensionInterface
 
         return $this->extension ??= new BundleExtension($this, $this->extensionAlias);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPath(): string
+    {
+        if (null === $this->path && false !== strrpos($path = parent::getPath(), \DIRECTORY_SEPARATOR.'src')) {
```

Comment:
perhaps assume the modern directory structure, aka `return \dirname(__DIR__);`

the we just flip the docs, to override getPath() if needed

Reply:
I agree. Let's make the default of `AbstractBundle` only support the new structure, and require bundles to override it when they want to stick to the old structure.

Reply:
Why not support both by default? this way, you will have to update the path only if you're using a weird convention.

Reply:
Reducing the amount of legacy code improves maintenance. Not just code, but especially knowledge. If two structures are supported, you'll have to know both structures in order to get working on a bundle. Removing a legacy structure (in the future) reduces cognitive overhead.

Besides that, this is claiming to support both as a "plug 'n play", but in reality it doesn't support most of the bundles that are actively maintained, but using the old structure for BC reasons. Almost all popular community bundles have migrated to a `src/` and `tests/` directory, with the legacy structure living in `src/`.

Reply:
I agree with this argument, especially from a maintenance pov: less "if"s the better :)

Reply:
Updated!